BDArmory/Control/BDModuleSurfaceAI.cs

```diff
@@ -690,12 +690,42 @@ void AttitudeControl(FlightCtrlState s)
                                 }
                             case WeaponClasses.Gun:
                                 {
-                                    if (weaponManager.currentTarget.isSplashed || ((weaponManager.currentTarget.isFlying || weaponManager.currentTarget.Vessel.situation == Vessel.Situations.LANDED) && weaponManager.currentGun.turret))
+                                    if (BDArmorySettings.BULLET_WATER_DRAG)
                                     {
-                                        if (Vector3.Distance(vessel.CoM, weaponManager.currentTarget.Vessel.CoM) > weaponManager.selectedWeapon.GetEngageRange())
-                                            targetAlt = 10; //come to periscope depth in preparation for surface attack when in range
-                                        else
-                                            targetAlt = -1;//in range, surface to engage with deck guns
+                                        if (weaponManager.currentTarget.isSplashed || ((weaponManager.currentTarget.isFlying || weaponManager.currentTarget.Vessel.situation == Vessel.Situations.LANDED) && weaponManager.currentGun.turret))
+                                        {
+                                            if (Vector3.Distance(vessel.CoM, weaponManager.currentTarget.Vessel.CoM) > weaponManager.selectedWeapon.GetEngageRange())
```

Comment:
I've converted the ones in SurfaceAI; converting the rest of the codebase will have to wait for a future commit, since there's already a lot of stuff in this one.